src/main/java/io/github/darealturtywurty/turtychemistry/core/MixinHooks.java

```diff
@@ -1,4 +1,18 @@
 package io.github.darealturtywurty.turtychemistry.core;
 
-public class MixinHooks {
+import io.github.darealturtywurty.turtychemistry.common.TurtyTags;
+import io.github.darealturtywurty.turtychemistry.core.init.ItemInit;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+public final class MixinHooks {
+
+
+    public static boolean isValidStackForAnvil(final ItemStack stack) {
+        return stack.is(Items.IRON_INGOT) || stack.is(TurtyTags.TURTY_ITEM_TAG_KEY) || stack.is(Items.COPPER_INGOT);
+    }
+
+    public static boolean isValidHammerForAnvil(final ItemStack stack) {
+        return stack.is(ItemInit.BASIC_HAMMER.get());
+    }
```

Comment:
I still don't think you are understanding what this is for.

Reply:
An example of what should be here is:
```java
public static void addItemToAnvil(BlockState pState, Level pLevel, BlockPos pPos, Player pPlayer, InteractionHand pHand, BlockHitResult pHit, CallbackInfoReturnable<InteractionResult> cir) {
        final Inventory playerInventory = pPlayer.getInventory();
        final ItemStack stack = pPlayer.getItemInHand(pHand);

        if (pPlayer.isCrouching() && pLevel.getBlockEntity(pPos) instanceof AnvilBlockEntity anvilBlockEntity) {
            if (!MixinHooks.isValidHammerForAnvil(stack)) {
                if (anvilBlockEntity.getItem().isEmpty()) {
                    if (MixinHooks.isValidStackForAnvil(stack)) {
                        anvilBlockEntity.setStackInSlot(stack.split(1));
                        anvilBlockEntity.setChanged();
                        cir.setReturnValue(InteractionResult.CONSUME);
                    }
                } else {
                    playerInventory.add(anvilBlockEntity.inventoryModule.getCapability().extractItem(0, 1, false));
                    cir.setReturnValue(InteractionResult.CONSUME);
                }
            } else {
                anvilBlockEntity.smithItem();
                if (stack.getDamageValue() < stack.getDamageValue()) {
                    stack.setDamageValue(stack.getDamageValue() + 1);
                } else {
                    stack.shrink(1);
                }
                cir.setReturnValue(InteractionResult.CONSUME);
            }
        }
}
```

Reply:
oooooh that makes more sense

Reply:
well, at least I think I get it, like all the logic is meant to be in the mixin hooks, and then the actual mixin just calls it in another method?

Reply:
Yes

Reply:
As stated:
> 1. The `MixinHooks` class is meant as a place to contain effectively static utility methods that the actual mixins will call. > This is useful in multiple ways:
> - It makes the mixins a lot more concise and therefore easier to read.
> - It allows much easier debugging of mixins as you can hotswap.
> - It keeps everything together a little more nicely.